core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -795,31 +795,40 @@ private[spark] class TaskSetManager(
       addPendingTask(index, readding = true)
     }
 
-    // Re-enqueue any tasks that ran on the failed executor if this is a shuffle map stage,
-    // and we are not using an external shuffle server which could serve the shuffle outputs.
-    // The reason is the next stage wouldn't be able to fetch the data from this dead executor
-    // so we would need to rerun these tasks on other executors.
-    if (tasks(0).isInstanceOf[ShuffleMapTask] && !env.blockManager.externalShuffleServiceEnabled) {
-      for ((tid, info) <- taskInfos if info.executorId == execId) {
-        val index = taskInfos(tid).index
-        if (successful(index)) {
-          successful(index) = false
-          copiesRunning(index) -= 1
-          tasksSuccessful -= 1
-          addPendingTask(index)
-          // Tell the DAGScheduler that this task was resubmitted so that it doesn't think our
-          // stage finishes when a total of tasks.size tasks finish.
-          sched.dagScheduler.taskEnded(tasks(index), Resubmitted, null, null, info, null)
+    reason match {
+      case LossReasonPending =>
+        // Nothing else to do until the actual loss reason is known.
```

Comment:
can you add that we expect `executorLost` to be called a second time with the real loss reason?